xray/plot/plot.py

```diff
@@ -424,6 +477,10 @@ def hist(self, ax=None, **kwargs):
     def line(self, *args, **kwargs):
         return line(self._da, *args, **kwargs)
 
+    @functools.wraps(plotter_1d)
+    def plotter_1d(self, *args, **kwargs):
```

Comment:
What about using this to define line plots and a few other plot methods that work automatically, e.g.,
```python
def line(self, *args, **kwargs):
    return plotter_1d(self._da, *args, plot_method='plot', **kwargs)

def bar(self, *args, **kwargs):
    return plotter_1d(self._da, *args, plot_method='bar', **kwargs)
```

That seems possibly a better idea that adding this generic method to the public API. Also, this would let us get ride of the redundant code for `.plot.line()``